corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowser.java

```diff
@@ -388,4 +392,107 @@ public void onError(Throwable throwable) {
 
         return streamDumper.getTxnRead().get();
     }
+
+    /**
+     * List all stream tags present in the Registry.
+     *
+     * @return stream tags
+     * */
+    public Set<String> listStreamTags() {
+        Set<String> streamTags = new HashSet<>();
+        runtime.getTableRegistry().getRegistryTable().values().forEach(record ->
+                streamTags.addAll(record.getMetadata().getTableOptions().getStreamTagList()));
+
+        log.info("\n======================\n");
+        log.info("Total unique stream tags: [{}]", streamTags.size());
+        streamTags.forEach(tag -> log.info(tag));
+        log.info("\n======================\n");
+
+        return streamTags;
+    }
+
+    /**
+     * List a map of stream tags to table names.
+     *
+     * @return map of tags to table names in the registry
+     */
+    public Map<String, List<TableName>> listTagsToTableMap() {
+        Map<String, List<TableName>> streamTagToTableNames = getMapTagToTableNames();
+        log.info("\n======================\n");
+        log.info("Total unique stream tags: [{}]", streamTagToTableNames.keySet().size());
+        log.info("Stream tags: {}\n", streamTagToTableNames.keySet());
+        streamTagToTableNames.forEach((tag, tableNames) -> printStreamTagMap(tag, tableNames));
+        log.info("\n======================\n");
+        return streamTagToTableNames;
+    }
+
+    /**
+     * List all tags for the given table.
+     *
+     * @param namespace namespace for the table of interest
+     * @param table table name of interest
+     */
+    public Set<String> listTagsForTable(String namespace, String table) {
+        Set<String> tags = new HashSet<>();
+        TableName tableName = TableName.newBuilder().setNamespace(namespace).setTableName(table).build();
+        CorfuRecord<CorfuStoreMetadata.TableDescriptors, CorfuStoreMetadata.TableMetadata> record = runtime.getTableRegistry()
+                .getRegistryTable().get(tableName);
+        if (record != null) {
+            tags.addAll(record.getMetadata().getTableOptions().getStreamTagList());
+            log.info("\n======================\n");
+            log.info("table: <{}${}> --- Tags[total={}] :: {}", namespace, table, tags.size(), tags);
+            log.info("\n======================\n");
+        } else {
+            log.warn("Invalid namespace {} and table name {}. Review or run operation --showTagsMap for complete map (all tables).", namespace, tableName);
+        }
+
+        return tags;
+    }
+
+    /**
+     * List all tables with a specific stream tag.
+     *
+     * @param streamTag specific stream tag, if empty or null return all stream tags map
+     * @return table names with given 'streamTag'
+     */
+    public List<TableName> listTablesForTag(@Nonnull String streamTag) {
+        if (streamTag == null || streamTag.isEmpty()) {
```

Comment:
Do we need log.warn() to indicate that the input streamTag is empty?
And is streamTag == null still necessary here with `@Nonnull`?

Reply:
Yes, the annotation is purely informational (for devs) but you can still pass in null values. Adding the log.warn()

Reply:
yes, the null check is not required

Reply:
The null check is required, if you experiment with it, you'll see you can do: browser.listTablesForTag(null); and removing it you would get a NPE.